app/src/main/java/com/team41/boromi/controllers/BookReturnController.java

```diff
@@ -0,0 +1,138 @@
+package com.team41.boromi.controllers;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.type.LatLng;
+import com.team41.boromi.callbacks.ReturnCallback;
+import com.team41.boromi.constants.CommonConstants.BookStatus;
+import com.team41.boromi.constants.CommonConstants.BookWorkflowStage;
+import com.team41.boromi.dbs.BookDB;
+import com.team41.boromi.dbs.BookReturnDB;
+import com.team41.boromi.models.Book;
+import com.team41.boromi.models.BookReturn;
+import com.team41.boromi.models.User;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.concurrent.Executor;
+import javax.inject.Inject;
+
+public class BookReturnController {
+
+  private final static String TAG = "ReturnController";
+  final Executor executor;
+  BookReturnDB returnDB;
+  BookDB bookDB;
+  FirebaseAuth auth;
```

Comment:
why do you need firebase auth

Reply:
Also annotate this controller with @Singleton so it is not recreated in each activity/fragment

Reply:
fixed!